src/odemis/driver/smaract.py

```diff
@@ -1215,6 +1215,7 @@ def __init__(self, name, role, locator, axes, ref_on_init=False, linear_speed=0.
                 raise model.HwError("Failed to find device, check it is connected and turned on")
             raise
         logging.debug("Successfully connected to SA_MC Controller ID %d", self._id.value)
+        self._deactive_pos_after_ref = None
```

Comment:
I meant the *opposite*. Keep the `pos_deactive_after_ref` boolean flag + the argument. However, you don't need to store the DEACTIVE value when the metadata changes, because it's already in the metadata.